### ISSUE TITLE
Feature preference to automatically remove dependencies

### ISSUE THREAD

**Issue by stevenwdv** [chocolatey/choco#2300]
It seems that currently we have to pass in `-x`/`--force-dependencies`/`--remove-dependencies`/equivalent to `choco uninstall` to remove package dependencies. I don't think I'd ever **not** want to uninstall dependencies when uninstalling a package, so I think it'd be very useful to have some feature preference to always uninstall dependencies, like:
```ps1
choco feature enable -n=alwaysUninstallDependencies
```
Maybe there should still be a way to choose to not uninstall dependencies.

Or am I understanding the option wrong and does `-x` also remove dependencies required by other packages and/or dependencies installed explicitly?

**Comment by pauby** [comment]
`-x` will remove dependencies if they are not required by other packages. 

As an example I installed `vlc` and `googlechrome` packages, both of which take a dependency on `chocolatey-core.extension`. I then uninstalled vlc with the `-x` argument.

```
PS C:\Users\WDAGUtilityAccount> choco uninstall vlc -x
Chocolatey v0.10.15
Uninstalling the following packages:
vlc
[NuGet] 'chocolatey-core.extension 1.3.5.1' skipped, because it is in use by 'GoogleChrome 91.0.4472.124'.

vlc v3.0.16
Uninstalling vlc...
0
vlc has been uninstalled.
 Running auto uninstaller...
 Skipping auto uninstaller - 'VLC media player' appears to have been uninstalled already by other means.
 vlc has been successfully uninstalled.

Chocolatey uninstalled 1/1 packages.
 See the log for details (C:\ProgramData\chocolatey\logs\chocolatey.log).
```

**Comment by stevenwdv** [comment]
@pauby Why is this tagged "Waiting on User"? Your comment just confirms that this would be a good idea.

**Comment by pauby** [comment]
Because I answered your question and was waiting for you to confirm you're happy with it.

With regards to this being a good idea, we already have the switch to allow dependencies to be uninstalled. This is the first request (that I remember) we've had for this so I have to question the effort versus benefit we'd get from adding something like this.

**Comment by stevenwdv** [comment]
@pauby Alright. Let me reverse the matter: why would one want (non-explicitly installed) dependencies to not be uninstalled?

**Comment by pauby** [comment]
* You may want to uninstall and reinstall the software and not want to have to reinstall the dependencies;
* The dependency may ba used by other software not managed by Chocolatey;
* The dependency may be software that you actually use directly;

That was just off the top of my head. There may be other reasons.

**Comment by stevenwdv** [comment]
@pauby You're right, there are indeed valid use cases. Especially the first one is something I hadn't thought about. However, in the other two cases, I think it would be better to explicitly install the package (does Chocolatey have support for this?).

What I'm saying is that, while there are valid use cases, in most cases I think that at least I would want to remove dependencies. The only thing that comes to mind that I'm not certain about is MSVC++ runtimes, because normally I don't explicitly install those; I just trust that they're updated automatically via Windows update (which I think is the case?), and I wouldn't want to remove those if other programs turn out to use them as well.
